### ISSUE TITLE
ExaBGP let users send routes without next-hop

### ISSUE THREAD

**Issue by dalasujal** [Exa-Networks/exabgp#468]
Hi..
I am trying to setup an ipv4 and ipv6 sessions and configured these in the same config file.
I am using 2 different processes for ipv4 and ipv6 neighbors.

The BGP sessions come up OK if I dont add the process to the config.

The BGP sessions come up OK if I dont add the "ipv4" process to the config.

But when I add the ipv4 process to the config, and as soon as a route announcement is made, the ipv6 session starts to flap. (the ipv6 process is not making any announcements)

This is the quagga (neighbor side) debug:

```
2016/08/22 02:54:00 BGP: xxxx::e8c9 OPEN has 4-octet AS number capability (65), length 4
2016/08/22 02:54:00 BGP: xxxx::e8c9 went from OpenSent to Deleted
2016/08/22 02:54:00 BGP: xxxx::e8c9 send message type 4, length (incl. header) 19
2016/08/22 02:54:00 BGP: xxxx::e8c9 went from OpenSent to OpenConfirm
2016/08/22 02:54:00 BGP: xxxx::e8c9 rcv message type 4, length (excl. header) 0
2016/08/22 02:54:00 BGP: xxxx::e8c9 went from OpenConfirm to Established
2016/08/22 02:54:00 BGP: xxxx::e8c9 send message type 4, length (incl. header) 19
2016/08/22 02:54:00 BGP: xxxx::e8c9 Missing well-known attribute 3.
2016/08/22 02:54:00 BGP: xxxx::e8c9 sending NOTIFICATION 3/3 (UPDATE Message Error/Missing Well-known Attribute) 1 bytes 03
2016/08/22 02:54:00 BGP: xxxx::e8c9 send message type 3, length (incl. header) 22
2016/08/22 02:54:00 BGP: Notification sent to neighbor xxxx::e8c9: type 3/3
2016/08/22 02:54:00 BGP: xxxx::e8c9 went from Established to Clearing
2016/08/22 02:54:00 BGP: xxxx::e8c9 went from Clearing to Idle
2016/08/22 02:54:15 BGP: xxxx::e8c9 went from Idle to Connect
2016/08/22 02:54:15 BGP: xxxx::e8c9 went from Connect to Active
2016/08/22 02:54:21 BGP: xxxx::e8c9 passive open
2016/08/22 02:54:21 BGP: xxxx::e8c9 went from Active to OpenSent
```

Below is the config file

```
group routers {
    router-id "local v4 addres";

    neighbor "neighbor ipv4" {
        local-address "local ipv4";
        local-as 65000;
        peer-as 65001;
    }

    family {
        ipv4 unicast;
    }

    process add-routes {
        run bin/route_announcer_start.sh iad 4;
    }
}


group routers_v6 {
    router-id "local v4 addres";

    neighbor "neighbor ipv6" {
        local-address "local ipv6 tunnel address";
        local-as 65000;
        peer-as 65001;
    }

    family {
        ipv6 unicast;
    }

    process add-routes-v6 {
        run bin/route_announcer_start.sh iad 6;
    }

```

**Comment by thomas-mangin** [comment]
Hello - could you please send me the output of exabgp when run with the '-d' option so I can see what is happening.

**Comment by dalasujal** [comment]
Hi Thomas,
Attached is the log file.


[exabgp-iad.log.2016-08-22-02-snippet.txt](https://github.com/Exa-Networks/exabgp/files/430921/exabgp-iad.log.2016-08-22-02-snippet.txt)

**Comment by thomas-mangin** [comment]
Can you please narrow the route to send to ONE to see what is not right. Quagga seems to say that one of the update sent is invalid: NOTIFICATION 3/3 (UPDATE Message Error/Missing Well-known Attribute) 1 bytes 03

One of the route sent looks like:
decoded update 1 10.10.10.0/24 origin igp as-path [ 65000 ] community 666:9999

So there is not next-hop set ... ExaBGP should really not accept that route as it is invalid - please set a next-hop on your route.

**Comment by dalasujal** [comment]
Hi Thomas, 
That is not the issue.
The route announcements are sent with next-hop self by the ipv4 process and works fine  for the ipv4 neighbor

What is happening is that for some reason, exa is also announcing the route (and changing something on  its way) for the ipv6 neighbor. The way its configured, the ipv6 neighbor should not even be getting the route announcement from its process.

**Comment by thomas-mangin** [comment]
There was no IPv6 in the log you gave me.

**Comment by dalasujal** [comment]
Hi Thomas,
There is no ipv6 route announcement because there was no announcement made.

May be I explained the issue incorrectly. let me try again this time with more detail. 

So, I have a config file(exabgp.ini - shown in my initial post) with 2 neighbors configured, and 2 separate processes start( for v4 and v6 each) once I load the exabgp.ini, it successfully brings up both the BGP sessions( v4 and v6)

Now, I want to announce a v4 route - I add that to my announce file, the v4 process picks it up and announces this to the v4 neighbor with next-hop self.
Nothing is done on the v6 end so far. No v6 route is added. v6 process is running but doing nothing.

But the moment I announced the v4 route to the v4 neighbor, the v4 neighbor gets the announcement but the v6 neighbor drops with the log messages as seen. 

Now as a work around, I just created 2 separate exabgp.ini for v4 and v6, and on announcing v4 routes, the v6 session does not drop. 

So I think its an issue with exabgp when there are 2 separate processes defined in the same exabgp.ini file.

**Comment by thomas-mangin** [comment]
ah, got it ! there is bugs on 3.4 (I can not fix without breakgin compatibility) causing cross-group API message passing.

Please create two configurations files and run exabgp with them both in command line.
It will fork two process and will do what you need.

**Comment by dalasujal** [comment]
Thanks for confirming this Thomas. I have created separate config files.
Is the newer version  that fixes this out already or is there an ETA?

**Comment by thomas-mangin** [comment]
The bug is fixed on master but the configuration format is not fixed yet.
